packages/core/core/src/RequestTracker.js

```diff
@@ -1543,55 +1548,94 @@ async function loadRequestGraph(options): Async<RequestGraph> {
 
   let cacheKey = getCacheKey(options);
   let requestGraphKey = `requestGraph-${cacheKey}`;
+  let timeout;
 
   if (await options.cache.hasLargeBlob(requestGraphKey)) {
-    let {requestGraph} = await readAndDeserializeRequestGraph(
-      options.cache,
-      requestGraphKey,
-      cacheKey,
-    );
+    try {
+      let {requestGraph} = await readAndDeserializeRequestGraph(
+        options.cache,
+        requestGraphKey,
+        cacheKey,
+      );
 
-    let opts = getWatcherOptions(options);
-    let snapshotKey = `snapshot-${cacheKey}`;
-    let snapshotPath = path.join(options.cacheDir, snapshotKey + '.txt');
+      let opts = getWatcherOptions(options);
+      let snapshotKey = `snapshot-${cacheKey}`;
+      let snapshotPath = path.join(options.cacheDir, snapshotKey + '.txt');
 
-    let timeout = setTimeout(() => {
-      logger.warn({
+      timeout = setTimeout(() => {
+        logger.warn({
+          origin: '@parcel/core',
+          message: `Retrieving file system events since last build...\nThis can take upto a minute after branch changes or npm/yarn installs.`,
+        });
+      }, 5000);
+      let startTime = Date.now();
+      let events = await options.inputFS.getEventsSince(
+        options.watchDir,
+        snapshotPath,
+        opts,
+      );
+      clearTimeout(timeout);
+
+      logger.verbose({
         origin: '@parcel/core',
-        message: `Retrieving file system events since last build...\nThis can take upto a minute after branch changes or npm/yarn installs.`,
+        message: `File system event count: ${events.length}`,
+        meta: {
+          trackableEvent: 'watcher_events_count',
+          watcherEventCount: events.length,
+          duration: Date.now() - startTime,
+        },
       });
-    }, 5000);
-    let startTime = Date.now();
-    let events = await options.inputFS.getEventsSince(
-      options.watchDir,
-      snapshotPath,
-      opts,
-    );
-    clearTimeout(timeout);
 
-    logger.verbose({
-      origin: '@parcel/core',
-      message: `File system event count: ${events.length}`,
-      meta: {
-        trackableEvent: 'watcher_events_count',
-        watcherEventCount: events.length,
-        duration: Date.now() - startTime,
-      },
-    });
+      requestGraph.invalidateUnpredictableNodes();
+      requestGraph.invalidateOnBuildNodes();
+      requestGraph.invalidateEnvNodes(options.env);
+      requestGraph.invalidateOptionNodes(options);
 
-    requestGraph.invalidateUnpredictableNodes();
-    requestGraph.invalidateOnBuildNodes();
-    requestGraph.invalidateEnvNodes(options.env);
-    requestGraph.invalidateOptionNodes(options);
-
-    try {
       await requestGraph.respondToFSEvents(
         options.unstableFileInvalidations || events,
         options,
         10000,
       );
       return requestGraph;
     } catch (e) {
+      if (timeout != null) {
+        clearTimeout(timeout);
+      }
+      let additionalMessage;
```

Comment:
you can extract this to a function but maybe it can be simpler (avoid the guessing snapshot path part & just log more lines rather than assigning to a variable etc.)